prog/dftb+/lib_elecsolvers/elecsolvers.F90

```diff
@@ -47,6 +47,19 @@ module dftbp_elecsolvers
     !> Whether the solver provides eigenvalues
     logical, public :: providesEigenvals
 
+    !> Whether the solver provides a band energy (sum of the eigenvalues)
+    logical, public :: providesBandEnergy
+
+    !> Whether the solver provides the TS term for electrons
+    logical, public :: providesElectronEntropy
+
+    !> Whether the solver provides electronic free energy (or is consistent with its evaluation)
+    logical, public :: providesFreeEnergy
+
+    !> Whether the solver provides the electron chemical potential (or is consistent with its
+    !> evaluation or it being suplied externally)
+    logical, public :: elecChemPotAvailable
```

Comment:
Why not `providesElecChemPot`?

Reply:
That name would be slightly misleading, as when its true the chemical potential may either be
1. actually provided by the solver (libOMM, NTPoly, PEXSI, ...)
2. specified by using the solver output combined with the required number of electrons (eigenvalues returned and we work out the fillings)
3. specified by an external bath/contact and then required by the solver (NEGF, ...)